### DISCUSSION TITLE
https://1kevinson.com/how-to-write-integration-tests-with-h2-in-memory-database-and-springboot/

### DISCUSSION THREAD

**Discussion by giscus** [1kevinson/BLOG-TUTORIALS#10] [Announcements]
# https://1kevinson.com/how-to-write-integration-tests-with-h2-in-memory-database-and-springboot/

H2 is an embedded, open-source, and in-memory database. It is a relational database management system written in Java. It is a client/server app

https://1kevinson.com/how-to-write-integration-tests-with-h2-in-memory-database-and-springboot/

**Comment by amin-tahri** [comment]
Hi, good article, thanks !

However the source code cannot be read on your github, a lot of configuration files are empty .icloud files

Regards

**Reply by 1kevinson** [reply]
Hello @amin-tahri thank you for this alert, I fixed the repos. Let me know if you face others issues. _Regards_

**Comment by Petrovich-A** [comment]
pleasure paper, however how to connect to H2 mem DB? Test connection is succeeded, but I don't see the tables while debug. (only queries in log)

**Reply by 1kevinson** [reply]
You cannot connect to H2 Database because it's configured only for the test purpose ([pay attention here  👀](https://1kevinson.com/how-to-write-integration-tests-with-h2-in-memory-database-and-springboot/#configuration)), so Hibernate launches the database (in the background), inserts the data, runs the tests and shutdown the database.

On the other hand, If H2 was used for a production code (i.e. configured in the `application.yml` file), then now you will be able to connect and see the tables where your data was inserted.

**Reply by Petrovich-A** [reply]
I have the same application-test.yml to configure DB. Can I connect and see the tables was inserted H2 only if using configuration file application.yml file?

**Reply by 1kevinson** [reply]
Yes, because the application.yml is for your production code.
So you have to enable that , via this configuration.

```yaml
spring:
  h2:
    console.enabled: true
```

Then start your application, and can navigate to http://localhost:8080/h2-console
_Change 8080 with your port_
<br>


🦉Checkout these links:
- [Accessing the H2 Console](https://www.baeldung.com/spring-boot-h2-database#h2-console)
- [How to connect and configure an in-memory H2 database in Spring Boot](https://www.youtube.com/watch?v=PSrHcCwvfVQ)

**Reply by Petrovich-A** [reply]
My purpose is to see the tables in H2 in memory for test. How can I do this?

**Reply by 1kevinson** [reply]
you cannot
